gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsJmxMonitor.java

```diff
@@ -0,0 +1,153 @@
+        }
+
+        // Set additional fields
+        clusterStats.proxyTo(backend.getProxyTo())
+                .externalUrl(backend.getExternalUrl())
+                .routingGroup(backend.getRoutingGroup());
+
+        ClusterStats stats = clusterStats.build();
+        log.debug("Completed monitoring for backend: %s. Stats: %s", backend.getProxyTo(), stats);
+        return stats;
+    }
+
+    private void processDiscoveryNodeManagerStats(String response, ClusterStats.Builder clusterStats)
+    {
+        try {
+            JsonNode rootNode = new ObjectMapper().readTree(response);
+            JsonNode attributes = rootNode.get("attributes");
+            if (attributes.isArray()) {
+                for (JsonNode attribute : attributes) {
+                    if ("ActiveNodeCount".equals(attribute.get("name").asText())) {
+                        int activeNodes = attribute.get("value").asInt();
+                        clusterStats.numWorkerNodes(activeNodes)
+                                .healthy(activeNodes - 1 > 0);
+                        log.debug("Processed DiscoveryNodeManager: ActiveNodeCount = %d, Health = %s",
+                                activeNodes, activeNodes - 1 > 0 ? "Healthy" : "Unhealthy");
+                        break;
+                    }
+                }
+            }
+        }
+        catch (Exception e) {
+            log.error(e, "Error parsing DiscoveryNodeManager stats");
+        }
+    }
+
+    private void processQueryManagerStats(String response, ClusterStats.Builder clusterStats)
+    {
+        try {
+            JsonNode rootNode = new ObjectMapper().readTree(response);
+            JsonNode attributes = rootNode.get("attributes");
+            if (attributes.isArray()) {
+                int queuedQueries = 0;
+                int runningQueries = 0;
+                for (JsonNode attribute : attributes) {
+                    String name = attribute.get("name").asText();
+                    if ("QueuedQueries".equals(name)) {
+                        queuedQueries = attribute.get("value").asInt();
+                    }
+                    else if ("RunningQueries".equals(name)) {
+                        runningQueries = attribute.get("value").asInt();
+                    }
+                }
+                clusterStats.queuedQueryCount(queuedQueries).runningQueryCount(runningQueries);
+                log.debug("Processed QueryManager: QueuedQueries = %d, RunningQueries = %d",
+                        queuedQueries, runningQueries);
+            }
+        }
+        catch (Exception e) {
+            log.error(e, "Error parsing QueryManager stats");
+        }
+    }
+
+    private String queryJmx(ProxyBackendConfiguration backend, String mbeanName)
+    {
+        String jmxUrl = backend.getProxyTo() + JMX_PATH + "/" + mbeanName;
+        log.debug("Querying JMX at URL: %s", jmxUrl);
+        OkHttpClient client = new OkHttpClient.Builder().build();
```

Comment:
Why not use airlift HTTP client?

Reply:
I tried this, but couldn't figure out how to use airlift HTTP client with Basic Auth using username and password.

Reply:
There is a [BasicAuthRequestFilter](https://github.com/airlift/airlift/blob/master/http-client/src/main/java/io/airlift/http/client/BasicAuthRequestFilter.java) class.

Reply:
Using airlift HTTP client now